### PR TITLE
update new RANSAC

### PR DESCRIPTION
**Pull request by ivashmak**
Related work https://github.com/opencv/opencv/pull/17683
### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [x] The PR is proposed to proper branch
- [x] There is reference to original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

**Comment by asmorkalov** [comment]
@vpisarev Please review the patch.

**Review by vpisarev — APPROVED** [review]

**Comment by vpisarev** [comment]
@alalek, if we still have time before code freeze for 4.5, please, merge it

File: modules/calib3d/src/usac/gamma_values.hpp [hunk 1]
```diff
@@ -1,237 +1,120 @@
...
+constexpr double stored_complete_gamma_values_n2[] = {1.77245,1.7045,1.67638,1.65484,1.63669,1.62073,1.60631,1.59307,1.58075,1.5692,1.55829,1.54792,1.53803,1.52855,1.51944,1.51066,1.50218,1.49397,1.48601,1.47827,1.47074,1.46341,1.45626,1.44927,1.44245,1.43577,1.42924,1.42283,1.41655,1.41039,1.40434,1.3984,1.39256,1.38682,1.38117,1.37561,1.37013,1.36474,1.35942,1.35418,1.34901,1.34392,1.33889,1.33392,1.32902,1.32418,1.3194,1.31467,1.31001,1.30539,1.30083,1.29631,1.29185,1.28744,1.28307,1.27874,1.27446,1.27023,1.26603,1.26188,1.25777,1.25369,1.24965,1.24565,1.24169,1.23776,1.23387,1.23001,1.22618,1.22238,1.21862,1.21489,1.21119,1.20752,1.20387,1.20026,1.19668,1.19312,1.18959,1.18608,1.18261,1.17915,1.17573,1.17232,1.16895,1.16559,1.16226,1.15896,1.15567,1.15241,1.14917,1.14595,1.14275,1.13958,1.13642,1.13329,1.13017,1.12708,1.124,1.12094,
```
Comment (alalek) [inline modules/calib3d/src/usac/gamma_values.hpp:9]: Please don't explode binaries or OpenCV source code with such blobs.

Please use on-demand one-time initializer of such tables.

Comment (ivashmak) [reply modules/calib3d/src/usac/gamma_values.hpp]: fixed it

Comment (alalek) [reply modules/calib3d/src/usac/gamma_values.hpp]: > on-demand
> one-time initializer

possible implementation is a singleton: https://en.wikipedia.org/wiki/Singleton_pattern

**Pull request merged**